src/main/java/mate/academy/hibernate/relations/Main.java

```diff
@@ -16,17 +19,17 @@ public static void main(String[] args) {
         SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
 
         Country usa = new Country("USA");
-        CountryService countryService = null; // TODO: initialize this instance
+        CountryService countryService = new CountryServiceImpl();
```

Comment:
You are instantiating `CountryServiceImpl` without passing the `SessionFactory`. You should pass the `SessionFactory` to the service implementation constructors to ensure that they use the same session factory.